### ISSUE TITLE
Resolve-MSBuild does not return the latest version installed.

### ISSUE THREAD

**Issue by lanierhall** [nightroman/Invoke-Build#122]
As best as I can share with you (since I don't have access to the server) we installed a combination of visual studio professional and visual studio build tools that doesn't return the latest version of MSBuild.exe

To recreate, on a new system, install an old version of Visual studio such as Visual studio professional 2017 version 1. Then install the latest version of Visual Studio Build Tools.

This will install MSBuild.exe in the following locations:
C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\MSBuild\15.0\Bin\MSBuild.exe
1) C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\MSBuild\15.0\Bin\MSBuild.exe
2) C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\MSBuild\15.0\Bin\amd64\MSBuild.exe

Resolve-MSBuild should return the BuildTools folder MSBuild.exe because it would be a later version, but it returns the older Professional version.

Nuget.exe selects the latest MSBuild.exe correctly with the same configured environment.

**Comment by nightroman** [comment]
> Nuget.exe selects the latest MSBuild.exe correctly with the same configured environment.

Can you describe this in more details?

**Comment by nightroman** [comment]
Do you have VSSetup module installed? If yes, can you get what you need using its parameters? I.e.

    Get-VSSetupInstance ??? | Select-VSSetupInstance ???

I am not suggesting to use VSSetup and this way instead of IB. I am thinking of some similar parameters to provide in IB, so that you can get what you need in the same way.

BTW, what is your command exactly?

**Comment by nightroman** [comment]
Could you please run the command

    Get-VSSetupInstance | Select-VSSetupInstance -Latest -Version 15.0 -Require Microsoft.Component.MSBuild

and see if it gets what you need?

If it does then we should think of adding the switch `Latest` to `Resolve-MSBuild`.

**Comment by lanierhall** [comment]
I ended up just writing some custom powershell logic:

```
$msBuildLocations = gci -Path "C:\Program Files (x86)\Microsoft Visual Studio\2017\**\MSBuild.exe" -Recurse 
$msBuildFileInfo = $msBuildLocations | select -ExpandProperty VersionInfo | Sort-Object -Property FileVersion -Descending | select -first 1

Set-Alias MSBuild ($msBuildFileInfo.FileName)
```

**Comment by nightroman** [comment]
Good workaround.
Please tell what you meant by "Nuget.exe selects the latest MSBuild.exe correctly".

**Comment by nightroman** [comment]
Meanwhile, I am going to add the switch `-Latest` and perform the similar sorting step if it is specified.

**Comment by lanierhall** [comment]
From my build output:

> 
> Task /./Restore
> 
> Restoring Nuget Packages
> 
> MSBuild auto-detection: using msbuild version '15.5.180.51428' from 'C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\MSBuild\15.0\bin'.
> Restoring packages for C:\...
>

**Comment by nightroman** [comment]
What is the command for restore? I am sorry, I am not using NuGet.exe (I prefer paket), could you be more specific, please?

**Comment by lanierhall** [comment]
nuget.exe -restore maybe

It doesn't matter, the point is that Invoke-Build's MSBuild.exe detection does not find the latest version while other's do.

**Comment by nightroman** [comment]
Well, it matters for me in order to learn what they do, why, find discussions, issues, etc.

Anyway, the "fix" is coming, the new switch `Latest`. By default, I think, it still makes sense to select by products: Enterprise, Professional, Community, others, in this order.

**Comment by nightroman** [comment]
Please try new switch in the released Invoke-Build 5.2.1
